docs/cli/graphql-transformer/directives.md

```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
 - [`translateText` Supported Language Codes](https://docs.aws.amazon.com/translate/latest/dg/what-is.html#what-is-languages)
 - [`convertTextToSpeech` Supported Voice IDs](https://docs.aws.amazon.com/polly/latest/dg/voicelist.html)
 
+## @http
+
+The `@http` directive allows you to quickly configure HTTP resolvers within your AWS AppSync API.
+
+### Definition
+
+```graphql
+directive @http(method: HttpMethod, url: String!, headers: [HttpHeader]) on FIELD_DEFINITION
+enum HttpMethod { PUT POST GET DELETE PATCH }
+input HttpHeader {
+  key: String
+  value: String
+}
+```
+
+### Usage
+
+The `@http` directive allows you to quickly connect HTTP or HTTPS endpoint to an AppSync API by creating an AWS AppSync HTTP resolver. To connect an AWS Lambda resolver, add the `@http` directive to a field in your `schema.graphql`. For example:
+
+```graphql
+type Query {
+  listPosts: Post @http(url: "https://www.example.com/posts")
+}
+```
+
+**Request Headers**
+
+The `@http` directive generates resolvers that can handle xml and json responses. If an HTTP method is not defined, `GET` is used by default. You can specify a list of static headers to be passed with the HTTP requests to your backend in your directive definition.
+
+```graphql
+type Query {
+  listPosts: Post @http(
+    url: "https://www.example.com/posts"
+    headers: [{key: "X-Header", value: "X-Header-Value"}]
+    )
+}
+```
+
+**Path Parameters**
+
+You can create dynamic paths by specifying params in the directive URL by using the special `:<param>` notation.
+
+Given the definition
+
+```graphql
+  getPost: Post @http(
+    url: "https://www.example.com/posts/:id"
+  )
+}
+```
+
+you can fetch a specific post
+
+```graphql
+query post {
+  getPost(params: {id: "POST_ID"}) {
+    id
+  }
+}
+```
+
+which will send
+
+```
+GET /posts/POST_ID
+Host: www.example.com
+``` 
+
+**Query String**
+
+You can send a query string with your request by specifying variables for your query.
+
+Given the definition
+
+```graphql
+type Query {
+  listPosts(sort: String!, from: String!, limit: Int!): Post @http(
+    url: "https://www.example.com/posts"
+    )
+```
+
+you can query for posts with
+
+```graphql
+query posts{
+  listPosts(query: {sort: "DESC", from: "last-week", limit: 5}) {
+    id
+    title
+    description
+  }
+}
+```
+
+which would send the following request:
+
+```
+GET /posts?sort=DESC&from=last-week&limit=5
+Host: www.example.com
+```
+
+**Request Body**
+
+The `@http` directive also allows you to specify the body of a request, which is used for `POST`, `PUT`, and `PATCH` requests. To create a new post, you can define the following.
```

Comment:
We should call out that ```query``` is also generated in the resulting schema, that enables the usage of the same fields in query params as in body. Another difference in behavior is that when something is defined as an url parameter that will NOT be added to query or body. Perhaps this should be called out in an earlier section or in the end in a summary section?

Reply:
adding